build.zig

```diff
@@ -40,5 +38,12 @@ pub fn build(b: *std.Build) void {
     tests.root_module.addImport("zigthesis", zigthesis_module);
     tests.root_module.addImport("utils", utils_module);
 
+    const internal_tests = b.addTest(.{
+        .root_source_file = b.path("src/zigthesis.zig"),
+        .target = target,
+        .optimize = optimize,
+    });
+
     test_step.dependOn(&b.addRunArtifact(tests).step);
+    test_step.dependOn(&b.addRunArtifact(internal_tests).step);
```

Comment:
It looks like the tests under `tests/` are good for testing the interface from the outside, but I wanted to write a test for the non-exported shrinking function directly.